joblib/memory.py

```diff
@@ -131,10 +131,20 @@ def _load_output(output_dir, func_name, timestamp=None, metadata=None,
                     t, str(signature), output_dir))
 
     filename = os.path.join(output_dir, 'output.pkl')
-    if not os.path.isfile(filename):
+
+    n_attempts = 3
+    sleep_time = 0.1
+
+    for _ in range(n_attempts):
```

Comment:
I certainly don't see the harm in making the sleep interval shorter (or start shorter and increase polling interval).

Cases we need to consider:
1. directory exists, PID file doesn't exist
    a. broken cache
    b. concurrency issue: can wait and go to (2b)
2. directory exists, PID file does exist
    a. broken cache
    b. concurrency issue (process with PID alive is a necessary but sufficient condition for this case)
        i. file moved within `eps` time
        ii. file not moved within `eps` time

I don't have any easy answers yet